android-exoplayer/src/main/java/com/brentvatne/exoplayer/ExoPlayerCache.java

```diff
@@ -91,7 +92,96 @@ public void run() {
                     );
 
                     if (!targetFile.exists()) {
-                        throw new Exception("Target file not present after writing bytes");
+                        throw new Exception("Target file bytes (" + bytesRead + ") not present after writing `" + targetFile.getPath() + "`");
+                    }
+
+                    Log.d(getName(), "Export succeeded");
+                    Log.d(getName(), targetFile.getPath());
+
+                    WritableMap result =  Arguments.createMap();
+                    result.putString("path", targetFile.getPath());
+                    result.putDouble("bytesRead", inputStream.bytesRead());
+                    result.putDouble("bytesWritten", targetFile.length());
+
+                    promise.resolve(result);
+                } catch (Exception e) {
+                    Log.d(getName(), "Export error");
+                    e.printStackTrace();
+
+                    String className = e.getClass().getSimpleName();
+                    promise.reject(className, className + ": " + e.getMessage());
+                    return;
+                }
+            }
+        }, "export_thread");
+        exportThread.start();
+    }
+
+    @ReactMethod
+    public void exportVideoImproved(final String url, final Promise promise) {
+        Log.d(getName(), "exportVideoImproved");
+
+        Thread exportThread = new Thread(new Runnable() {
+            @Override
+            public void run() {
+                Log.d(getName(), "Exporting...");
+                Log.d(getName(), url);
+                final Uri uri = Uri.parse(url);
+                final DataSpec dataSpec = new DataSpec(uri, 0, C.LENGTH_UNSET, null);
+                final SimpleCache downloadCache = ExoPlayerCache.getInstance(getReactApplicationContext());
+                CacheKeyFactory cacheKeyFactory = ds -> CACHE_KEY_PREFIX + "." + CacheUtil.generateKey(ds.uri);;
```

Comment:
I'm not too sure about writing Java code but is the double `;` at the end of the line wanted?